iothub/device/src/Transport/RetryDelegatingHandler.cs

```diff
@@ -469,11 +462,19 @@ private async Task EnsureOpenedAsync(CancellationToken cancellationToken)
                 {
                     if (Logging.IsEnabled) Logging.Info(this, "Opening connection", nameof(EnsureOpenedAsync));
                     await OpenAsyncInternal(cancellationToken).ConfigureAwait(false);
-                    _opened = true;
-                    _openCalled = true;
+                    if (!_disposed)
```

Comment:
I don't understand this code:
1. The check for disposed is already done in the base class.
2. Dispose isn't acquiring this lock so the `_disposed` variable isn't interlocked between this and `Dispose()`.

Reply:
1. didn't handle race condition
2. `volatile bool _disposed`

Reply:
point 1 is to resolve race condition. point 2 is resolved by volatile flag on _dispose


---
In reply to: [303071138](https://github.com/Azure/azure-iot-sdk-csharp/pull/1001#discussion_r303071138) [](ancestors = 303071138)

Reply:
This is by design

Reply:
@prmathur-microsoft please explain why this is by design: the code  opened the connection and now it stops handling the disconnect. The retry handler is common for amqp and mqtt and the current design clearly specifies this needs to be done.